ecs-cli/modules/cli/regcreds/create_task_execution_role.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2015-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package regcreds
+
+import (
+	"fmt"
+	"time"
+
+	iamClient "github.com/aws/amazon-ecs-cli/ecs-cli/modules/clients/aws/iam"
+	kmsClient "github.com/aws/amazon-ecs-cli/ecs-cli/modules/clients/aws/kms"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/utils"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iam"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	assumeRolePolicyDocString = `{"Version":"2008-10-17","Statement":[{"Sid":"","Effect":"Allow","Principal":{"Service":"ecs-tasks.amazonaws.com"},"Action":"sts:AssumeRole"}]}`
+	roleDescriptionString     = "Role generated by the ecs-cli"
+)
+
+type executionRoleParams struct {
+	CredEntries map[string]CredsOutputEntry
+	RoleName    string
+	Region      string
+}
+
+func createTaskExecutionRole(params executionRoleParams, iamClient iamClient.Client, kmsClient kmsClient.Client) error {
+	log.Infof("Creating resources for task execution role %s...", params.RoleName)
+
+	// generate policy document
+	policyDoc, err := generateSecretsPolicy(params.CredEntries, kmsClient)
```

Comment:
nit: I think this call should come after `createOrFindRole` and before `createRegistryCredentialsPolicy`